src/eko/output/struct.py

```diff
@@ -650,42 +603,61 @@ def bootstrap(dirpath: os.PathLike, theory: dict, operator: dict):
         (dirpath / PARTSDIR).mkdir()
         (dirpath / OPERATORSDIR).mkdir()
 
-    @staticmethod
-    def extract(path: os.PathLike, filename: str) -> str:
-        """Extract file from disk assets.
-
-        Note
-        ----
-        At the moment, it only support text files (since it is returning the
-        content as a string)
+    @classmethod
+    def open_tar(cls, path: os.PathLike):
+        """Extract EKO into a temporary folder.
 
         Parameters
         ----------
-        path: os.PathLike
-            path to the disk dump
-        filename: str
-            relative path inside the archive of the file to extract
-
-        Returns
-        -------
-        str
-            file content
-
+        path : os.PathLike
+            EKO file name
         """
+        if not tarfile.is_tarfile(path):
+            raise OutputNotTar
         path = pathlib.Path(path)
 
-        with tarfile.open(path, "r") as tar:
-            fd = tar.extractfile(filename)
-            content = fd.read().decode()
+        # dump to temporary directory
+        target = tempfile.mkdtemp()
+        target = pathlib.Path(target)
+        shutil.unpack_archive(path, target, "tar")
+
+        # load necessary assets
+        theory = yaml.safe_load((target / THEORYFILE).read_text())
+        operator = yaml.safe_load((target / OPERATORFILE).read_text())
+
+        eko = cls.detached(theory, operator, path=path, working_dir=target)
+        logger.info(f"Operator loaded from path '{path}' into '{target}'")
+        return eko
+
+    def write_tar(self):
+        """Write the operator back to disk."""
+        shutil.make_archive(self.path.stem, "tar", self.working_dir)
+        shutil.move(self.path.name, self.path)
 
-        return content
+    @classmethod
+    @contextlib.contextmanager
+    def open(cls, path: os.PathLike):
+        """Open the EKO object for reading.
+
+        Parameters
+        ----------
+        path : os.PathLike
+            EKO file name
+        """
+        obj = None
+        try:
+            obj = cls.open_tar(path)
+            yield obj
+        finally:
+            obj.write_tar()
+            shutil.rmtree(obj.working_dir)
```

Comment:
Move this to a separate `close()` function, such that is available also when the context manager is not used.